Fcom 100 file info approach

karoraita opened the pull request:
Move feature:

- Added bool DeleteSourceFile to enable source file removal after copy.
- Better Path handling.
- More informative error messages.
- Created helper method to allow overwrite feature to work on move (SDK does not support it natively)
- Updated readme.

Tested @ Grayskull.

karoraita committed:
Fixed path of "Building from source"

karoraita committed:
Fixed error message, some minor Resharper adjustments to control flow.

karoraita committed:
Added option to move files in Download task. Incremented version.

karoraita committed:
Inverted if.

karoraita committed:
Inverted if.

karoraita committed:
Better error messages. Ternary and fileinfo for result.

karoraita committed:
Updated readme.

karoraita committed:
AWS SDK MoveToLocal did not have overwrite option, so I created something like it.

karoraita committed:
Added Combine method for path handling.

karoraita committed:
Incremented version.

karoraita committed:
Rethrow exception with filename info in error message.

karoraita committed:
Switched around trycatch to accommodate both Move and Copy

karoraita committed:
Wrong exception.

karoraita committed:
Incremented version.

karoraita committed:
Updated readme.

karoraita committed:
Removed Frends.Task.Attributes, using DataAnnotations instead.

karoraita committed:
Removed remaining references to Frends.Task.Attributes

juralaakkonen reviewed (changes requested):
Taski luokkien nimet olisi hyvä olla samat kuin tiedostojen nimet,
Esim DownloadTask.cs tiedostossa, luokan nimi on vain Download. Pitäisi olla DownloadTask. Sama muissa.

Download taski:
Hakemisto ja tiedosto yhdistetään nyt: var path = input.DestinationPath + file.Name; Parempi olisi käyttää Path.Combine(input.DestinationPath, file.Name) - tämä on huomattavasti parempi (ei esim välitä jos uupuu keno välistä)

karoraita committed:
Renamed classes to match filenames.
Removed unnecessary using statement.
Added ConfigHandler for future tests.
Minor tweaks.

karoraita committed:
Resharper prettify

karoraita committed:
Should not use path in bucketname anymore.
